src/ds1302.c

```diff
@@ -2,130 +2,122 @@
 // http://datasheets.maximintegrated.com/en/ds/DS1302.pdf
 //
 
-#pragma callee_saves ds_writebit,ds_readbit
+#pragma callee_saves sendbyte,readbyte
 #pragma callee_saves ds_writebyte,ds_readbyte
 
 #include "ds1302.h"
 
 #define MAGIC_HI  0x5A
 #define MAGIC_LO  0xA5
 
-
-void ds_ram_config_init(uint8_t config[4]) {
+void ds_ram_config_init() {
     uint8_t i;
     // check magic bytes to see if ram has been written before
     if ( (ds_readbyte( DS_CMD_RAM >> 1 | 0x00) != MAGIC_LO || ds_readbyte( DS_CMD_RAM >> 1 | 0x01) != MAGIC_HI) ) {
         // if not, must init ram config to defaults
         ds_writebyte( DS_CMD_RAM >> 1 | 0x00, MAGIC_LO);
         ds_writebyte( DS_CMD_RAM >> 1 | 0x01, MAGIC_HI);
 
-        //for (i=0; i<4; i++)
-            //ds_writebyte( DS_CMD_RAM >> 1 | (i+2), 0x00);
-	ds_ram_config_write(config);
+	ds_ram_config_write();
 	return;
     }
     
     // read ram config
-    for (i=0; i<4; i++)
-        config[i] = ds_readbyte(DS_CMD_RAM >> 1 | (i+2));
+    for (i=0; i!=4; i++)
```

Comment:
nice. Looks like you are finding compiler optimizations that sdcc should be implementing... Have you filed any enhancement bugs with them? (I assume that the !=4 is the functional equivalent of <4 in a for loop that the compiler optimizes as it _should_ be never possible to overshoot this if the compiler is implementing the assembly ;)